src/components/CartItems.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux'
+import { ListGroup } from 'react-bootstrap';
+import { AppDispatch } from '../actions';
+import { AppState } from "../store";
+import { CartViewModifier } from '../types/cart';
+import { removeFromCart } from '../actions/cart';
+
+type Props = {
+  cartView?: CartViewModifier
+}
+
+const CartItems: React.FC<Props> = ({ cartView = CartViewModifier.NONE }) => {
+  const products = useSelector((state: AppState) => state.cartState.products);
+  const dispatch = useDispatch<AppDispatch>();
+  const cartTotal = products.reduce((total, prod) => total + prod.price, 0);
+  return (
+    <ListGroup>
+      { products.map((product, i) => (
+        <ListGroup.Item key={i}>
```

Comment:
Since this list could get resorted (based on the name sort), the `key={i}` can lead to weirdness:
https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

Might be time to create an ID for each product, or, if the title is unique, then use that.

Reply:
I've used `product.id` - thanks for the link.